tutorial.md

```diff
@@ -0,0 +1,1022 @@
+    but perhaps this will give people the start they need and then they
+    can focus on more important things than how their bar looks :)
+
+  * Is `xscreensaver` still relevant?
+
+
+# XMonad Configuration Tutorial
+
+I'm going to take you, step-by-step, through the process of configuring
+xmonad, setting up a status bar with [xmobar], setting up a tray with
+[trayer-srg], and making it all play nicely together.
+
+I assume that you have read the [xmonad guided tour] already.  It is a
+bit dated at this point but, because xmonad is stable, the guide should
+still give you a good overview of the most basic functionality.
+
+Before we begin, here are two screenshots of the kind of desktop we will
+be creating together.  In particular, a useful layout we'll conjure into
+being—the three columns layout from [XMonad.Layout.ThreeColumns] with
+the ability to magnify stack windows with [XMonad.Layout.Magnifier]:
+
+<p>
+<img alt="blank desktop" src="https://user-images.githubusercontent.com/50166980/111586872-c0eb0e00-87c1-11eb-9129-c2781bbbf227.png" width="550">
+<img alt="blank desktop" src="https://user-images.githubusercontent.com/50166980/111586885-c47e9500-87c1-11eb-8e8e-83d4d33c4c8d.png" width="550">
+</p>
+
+So let's get started!
+
+## Preliminaries
+
+First you'll want to install xmonad.  You can either do this with your
+systems package manager, or (a bit more advanced) via `stack`/`cabal`.
+You can find instructions for the latter in the [xmonad-testing]
+repository.  I'm going to assume xmonad version `0.16` and
+xmonad-contrib version `0.17` here, though most of these steps should
+work with older versions as well.  When we get to the relevant parts,
+will point you to alternatives that work with at least xmonad version
+`0.15` and xmonad-contrib version `0.16`.  This will usually be
+accompanied by a big fat "_IF YOU ARE ON A VERSION `< 0.17`_", so don't
+worry about missing it!
+
+Throughout the tutorial I will use, for keybindings, a syntax very akin
+to the [GNU/Emacs conventions] for the same thing—so `C-x` means "hold
+down the control key and then press the `x` key".  One exception is that
+in our case `M` will not necessarily mean Alt, but "your modifier key";
+this is Alt by default, although many people map it to Super instead (I
+will show you how to do this below).
+
+This guide will work for any GNU/Linux distribution and, provided that
+`stack` or `cabal` is used for installing xmonad (again see
+[xmonad-testing] for more information), even for BSD folks.  Because
+debian-based distributions are still rather popular, I will give you the
+`apt` commands when it comes to installing software.  If you use another
+distribution, just substitute the appropriate commands for your system.
+
+To install xmonad, as well as some utilities, via `apt` you can just run
+
+``` shell
+  apt-get install xmonad libghc-xmonad-contrib-dev libghc-xmonad-dev suckless-tools
```

Comment:
Thanks!  

Yeah I'd expect most people to not actually run debian but a derivative thereof, so this will _probably_ be fine.  Really what I'd like to do is to tell people to only install cabal and/or stack with their distros package manager and use projects to install anyting Haskell related; sadly, though, I can't see that happening anytime soon...

Reply:
Stumbled on https://wiki.haskell.org/Xmonad/Installing_xmonad which you might want to link here.

Reply:
That page looks a bit modly, especially debian's `libghc6-xmonad-contrib-dev` :)  Though it would be a cool idea to have all of these kinds of installation options on the new website and then link to that page!